Kent-frontend-storageSession-refactor

KentMayoya opened the pull request:
Things to test
- Navigating to /dashboard (without filling out a survey) displays no recommendations text
- Filling out the form will redirect the user to /dashboard and display recommendations regardless of whether the user is logged in or not
- If the user is logged in the answers will be pushed into supabase, if not, no data is pushed (but recommendations still display)

Note: Previous submissions is not displayed. Only the most recent recommendations are displayed.

KentMayoya committed:
sessionStorage stores questionId: answerOrder

KentMayoya committed:
Session storage now stores the question id, answer index, and answer content for each question

KentMayoya committed:
submit-answers api works

KentMayoya committed:
Recommendations is stored in sessionStorage

KentMayoya committed:
Recommendations now loaded from sessionStorage

KentMayoya committed:
Added case if there are no recommendations

KokonutGamer reviewed (changes requested):
## Summary
Although this implementation addresses the need for public users to be able to use the platform, the logic for handling the questionnaire breaks when a user does sign in.

## Core Issue
After taking the questionnaire, anonymous users (users not signed into the site) receive recommendations immediately from our service. This behaves just as we've intended.

![Screenshot 2025-06-06 015003](https://github.com/user-attachments/assets/4c7fa1f7-b9b6-4c4f-8fda-f4469f5a7982)

Once on the dashboard screen, a user may decide to sign in. They'll be redirected to the login page, and from there, redirected back to the dashboard on receiving their token.

![Screenshot 2025-06-06 015027](https://github.com/user-attachments/assets/adedf39e-9c1d-48e4-8099-379357b1e830)

Currently, **no behavior is enforced to ensure cached submissions are queried to the database**. This is the main issue with the current build which should be addressed before merging into main.

![Screenshot 2025-06-06 015205](https://github.com/user-attachments/assets/3c143e1a-3b21-48d3-954d-be1054a3557a)

## Recommended Approach
Update the recommendation cards on entering the dashboard with a authorized token. This might need more research, but you can potentially use `useState` or an existing React hook (perhaps `AuthContext`?) to implement this.

KentMayoya committed:
Moved API logic to useSurveySubmits and used localStorage to track if survey answers were submitted

KentMayoya commented:
I have updated the code. This is the behavior I have implemented:

User is not logged in. Take and submit the survey.
Recommendations appear on the dashboard.
Survey answers do not appear on supabase.
Sign in.
Survey answers now appear on supabase.
Sign out.
Navigate back to /dashboard.
No recommendations appear.

KokonutGamer reviewed (approved):
Works now for me!
